### DISCUSSION TITLE
Terminal version of Langfuse (Langfuse CLI)

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [langfuse/langfuse#3723] [Ideas]
### Describe the feature or potential improvement

Hello,
Thanks for providing observation and engineering tool for Opensource world.
Just as terminal junky person, wanted to know, if we have any terminal version of Langfuse or something?
Thanks

**Comment by marcklingen** [comment]
Interesting idea, there is no cli yet, let's track interest for this here

**Comment by Shahin-rmz** [comment]
Thanks for the reply,
I mean as the project has API architecture and it has front end Back end structure, it would be not that hard to implement CLI.
Other than that, if we want to have interactive Tui window, some libraries like textualise are very helpful !
I hope I can do something for the project.

**Comment by vicentereig** [comment]
I've put together a quick cli over here that I use with codex an claude. Happy hacking.

https://github.com/vicentereig/lf-cli

```
⏺ Looking at those 2 traces more closely:

  Trace 1: "optimization.miprov2_compile"
  - This is a DSPy MIPROv2 optimization run from your dspy-ruby project
  - It's optimizing a ReAct agent for HotPotQA question answering
  - Configuration:
    - 10 training examples, 5 validation examples
    - 6 trials in "greedy" mode (explores different prompts/examples)
    - Light mode (faster but less thorough)
  - 152 observations over 82.9 seconds
  - Cost: $0.0062
  - This is the optimizer trying different combinations of prompts and few-shot examples to improve the agent

  Trace 2: "evaluation.batch"
  - This is a batch evaluation of the DSPy::ReAct agent
  - Ran on 5 examples with 3 parallel threads
  - Had a metric to measure performance
  - 31 observations over 18 seconds
  - Cost: $0.0010
  - This likely ran before the optimization to get a baseline score

  So these traces tell the story: Evaluate baseline ReAct performance → Optimize with MIPROv2 → (presumably) Re-evaluate to see improvement. This is a
  typical DSPy optimization workflow for improving a question-answering agent.
```

**Reply by Tar-ive** [reply]
Thank you for creating this, still need to put this into actual deployment but looks really well documented, and usable.

**Reply by vicentereig** [reply]
good stuff - I've been using it from codex/claude [while running dpsy.rb optimizations](https://x.com/highwayvaquero/status/1983112028429627838) and I'm pretty happy. Hoping it works for you too!

**Comment by jkryanchou** [comment]
Awesome, and now langchain already provides the langsmith-fetch cli tools which could be invoked with Claude Code and deepagents. 

I suggested that the langfuse could do the same thing order to provide an methodology to level up the deepagents.

**Comment by ossa-ma** [comment]
+1 for a langfuse cli, would make scanning traces and prompt/eval management way easier with claude code.

**Reply by robinsadeghpour** [reply]
+1 here, developer experience with agentic tools like claude code would be awesome!

**Comment by nimarb** [comment]
hi @Shahin-rmz , @vicentereig , @jkryanchou , @robinsadeghpour , @Tar-ive , thank you all for the feedback!
We have now shipped a CLI, see : https://langfuse.com/changelog/2026-02-17-langfuse-cli

or install via `npm i -g langfuse-cli`

Happy to hear your feedback on this!
It uses the external API and thus has all capabilities. You can get an integrated skill via `langfuse get-skill` to pipe into your agent.

**Reply by jkryanchou** [reply]
Awesome Jobs~
